File: flink-job/src/main/java/com/predict/utils/HdfsUtils.java
package com.predict.utils;

import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.fs.FileSystem;
import org.apache.hadoop.fs.Path;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.URI;

public class HdfsUtils {
    private static final String HDFS_URI = "hdfs://master:9000"; // 根据实际HDFS地址调整

    public static String readFileAsString(String hdfsPath) throws Exception {
        Configuration conf = new Configuration();
        FileSystem fs = FileSystem.get(URI.create(HDFS_URI), conf);
        Path path = new Path(hdfsPath);
        if (!fs.exists(path)) {
            return null;
        }
        StringBuilder sb = new StringBuilder();
        try (BufferedReader br = new BufferedReader(new InputStreamReader(fs.open(path)))) {
            String line;
            while ((line = br.readLine()) != null) {
                sb.append(line);
            }
        }
        return sb.toString();
    }
}

File: spark-offline-job/src/main/scala/com/predict/utils/HdfsUtils.scala
package com.predict.utils

import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.fs.{FileSystem, Path}
import org.slf4j.LoggerFactory

object HdfsUtils {

  private val logger = LoggerFactory.getLogger(this.getClass)
  private val conf = new Configuration()
  conf.set("fs.defaultFS", "hdfs://master:9000")

  def getFileSystem: FileSystem = FileSystem.get(conf)

  def exists(pathStr: String): Boolean = {
    val fs = getFileSystem
    fs.exists(new Path(pathStr))
  }

  def deleteAndCreateDir(pathStr: String): Unit = {
    val fs = getFileSystem
    val path = new Path(pathStr)
    try {
      if (fs.exists(path)) {
        logger.info(s"Deleting existing path: $pathStr")
        fs.delete(path, true)
      }
      logger.info(s"Creating directory: $pathStr")
      fs.mkdirs(path)
    } catch {
      case e: Exception =>
        logger.error(s"Failed to delete/create directory: $pathStr", e)
        throw e
    }
  }
}
